go.mod

```diff
@@ -0,0 +1,20 @@
+module k8s.io/gengo
+
+go 1.13
+
+require (
+	fake v0.0.0
+	github.com/davecgh/go-spew v1.1.1
+	github.com/google/go-cmp v0.4.0
+	github.com/google/gofuzz v1.1.0
+	github.com/spf13/pflag v1.0.5
+	golang.org/x/tools v0.0.0-20200505023115-26f46d2f7ef8
+	k8s.io/klog/v2 v2.0.0
+	sigs.k8s.io/yaml v1.2.0
+)
+
+replace (
+	fake v0.0.0 => ./fake // vendor the fake dep
```

Comment:
this is definitely unexpected. if we have modules as test dependencies put them under a testdata folder or something similar

Reply:
testdata is not going to work with go mod from what I can tell. The original testing pattern is faking out go mod, or the testdata folder needs a go.mod file. I can try that.

Reply:
you would need to set the current working dir to the folder under testdata which defined the go.mod file

Reply:
I got it to work! but go mod added an indirect import, and I added a replace directive to allow it to find the sub-module. 

This should get trimmed off when something deps on this package.